content/posts/2024-11-05-flang-new.md

```diff
@@ -0,0 +1,963 @@
+  This effort was led by Arm, with Peter Waller (Arm) going so far as to write
+  a [custom tool](https://github.com/llvm/llvm-project/commit/55d5e6cbe2509a24132d056e1f361dc39312929b#diff-c389405236998090c7c8b9741506f01fb28abbd7da52e9566323c585ac0eb89cL910)
+  to rewrite the history of F18.
+
+  Kiran Chandramohan (Arm) [proposes](https://groups.google.com/a/tensorflow.org/g/mlir/c/SCerbBpoxng/m/bVqWTRY7BAAJ)
+  an OpenMP dialect for MLIR, with the intention of using it in Flang (discussion
+  continues on [Discourse](https://discourse.llvm.org/t/rfc-openmp-dialect-in-mlir/397)
+  during the following January).
+* **Februrary 2020** The [plan](https://discourse.llvm.org/t/plan-for-landing-flang-in-monorepo/54546)
+  for improvements to F18 to meet the standards required for inclusion in the
+  LLVM monorepo is announced by Richard Barton (Arm).
+* **April 2020** Upstreaming of F18 into the LLVM monorepo is
+  [completed](https://github.com/llvm/llvm-project/commit/b98ad941a40c96c841bceb171725c925500fce6c).
+
+At this point what was in the LLVM monorepo was F18, the rewritten frontend of
+Classic Flang. Classic Flang remained unchanged, still using the PGI based frontend.
+
+Around this time work started in the Classic Flang repo on the `fir-dev` branch
+that would enable code generation when using F18.
+
+For the following events remember that Classic Flang was still in use. The Classic
+Flang binary is named `flang`, just like the folder F18 now occupies in the LLVM monorepo.
+
+* **April 2020** The first attempt at adding a new compiler driver for Flang is
+  [posted](https://reviews.llvm.org/D79092) for review. It used the name
+  `flang-tmp`. This change was later abandoned in favour of a different approach.
+* **September 2020** Flang's new compiler driver is [added](https://reviews.llvm.org/D86089)
+  as an experimental option. This is the first appearance of the `flang-new` binary,
+  instead of `flang-tmp` as proposed before.
+
+  > The name was intended as temporary, but not the driver.
+  * Andrzej Warzyński (Arm, Flang Driver Maintainer)
+
+* **October 2020** Flang is included in an LLVM release for the first time in
+  LLVM 11.0.0. There was an `f18` binary and the aforementioned script
+  `flang.sh`.
+* **August 2021** `flang-new` is no longer experimental and [replaces](https://reviews.llvm.org/D105811)
+  the previous Flang compiler driver `f18`.
+* **October 2021** First appearance of the `flang-new` binary in an LLVM release
+ (13.0.0).
+* **April 2022** NVIDIA [ceases development](https://discourse.llvm.org/t/nvidia-transition-from-fir-dev/61947)
+  of the `fir-dev` branch in the Classic Flang project. Upstreaming of `fir-dev`
+  to the LLVM Project begins around this date.
+
+  `flang-new` can now do [code generation](https://reviews.llvm.org/D122008)
+  if the `-flang-experimental-exec` option is used. This change leveraged work
+  originally done on the `fir-dev` branch.
```

Comment:
It would be nice to note here (and below) which LLVM version the various milestones first appeared in.

Reply:
Yes good idea. I have added the significant releases as their own entries in the timeline.

It's good because now we can see that (at least from an external perspective) there was only a month between the first release with `flang-new` as the only Flang binary, and the decision to rename it to `flang`.